### PR TITLE
paramdigger: Handle session change & previous PR follow-up

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Add change note.
- CacheController > Remove unnecessary else block and conditional handling. (Mentioned in https://github.com/zaproxy/zap-extensions/pull/4561/files)
- ExtensionParamDigger > Add and use SessionChangedListener.
- HeaderGuesser > Make constant final. (Mentioned in https://github.com/zaproxy/zap-extensions/pull/4561/files)

## Related Issues
- zaproxy/zaproxy#7694
- zaproxy/zap-extensions#4561

## Checklist
- [NA] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [NA] Write tests
- [NA] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java [hunk 1]
```diff
@@ -21,17 +21,20 @@
...
+import org.parosproxy.paros.model.Session;
 import org.parosproxy.paros.network.HttpMessage;
 import org.zaproxy.addon.paramdigger.gui.ParamDiggerDialog;
 import org.zaproxy.addon.paramdigger.gui.ParamDiggerPanel;
 import org.zaproxy.addon.paramdigger.gui.PopupMenuParamDigger;
 import org.zaproxy.zap.utils.DisplayUtils;
 import org.zaproxy.zap.view.ZapMenuItem;
 
-public class ExtensionParamDigger extends ExtensionAdaptor {
+public class ExtensionParamDigger extends ExtensionAdaptor implements SessionChangedListener {
```
Comment (thc202) [inline addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java:37]: Should not be exposed through the extension.

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java [hunk 2]
```diff
@@ -861,20 +861,15 @@ private boolean checkAlwaysMiss(String url, Method method, Cache cache) {
...
-                    }
-                }
-                return false;
             }
+            if (!this.checkCacheHit(indicValue, cache) && cache.getIndicator() != null) {
+                sleeper(2000);
+                httpSender.sendAndReceive(msg);
+                addCacheMessage(msg);
+                indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
+                return this.checkCacheHit(indicValue, cache);
```
Comment (thc202) [inline addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java:870]: It was returning false when true, should have test.

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java (hunk 1 above)
Comment (kingthorin) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java:37]: On the actual panel then?

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java (hunk 2 above)
Comment (kingthorin) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java:870]: Will review, and see what I can do/add. I did double check that existing tests passed.

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java (hunk 1 above)
Comment (thc202) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java:37]: Here but an inner class.

File: addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java (hunk 2 above)
Comment (kingthorin) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java:870]: Agreed the logic was the other way around. However, I can't come up with a test case where it matters 😞

Comment (thc202) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java]: You mean based on the existing tests?

Comment (kingthorin) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java]: Yes, shouldNotFindCacheWithAnyCacheBuster seems to handle one case. But, really passes whether or not that block is true or false 🤷‍♂️

You can validate simply with:
```java
                System.out.println("Here");
                return false;//!this.checkCacheHit(indicValue, cache);
```
and
```java
                System.out.println("Here");
                return true;//!this.checkCacheHit(indicValue, cache);
```

Comment (kingthorin) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java]: Any thoughts?

Comment (thc202) [reply addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java]: I will have to check the code/tests.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**